extensions/vscode/src/utils/files.ts

```diff
@@ -267,3 +269,50 @@ export function relativeDir(uri: Uri): string | undefined {
   // Otherwise, return the relative path VSCode expects
   return workspace.asRelativePath(dirname);
 }
+
+/**
+ * Returns a VSCode workspace relative file path for a given URI.
+ * This will include the path to a file
+ *
+ * @param uri The URI to get the relative path of
+ * @returns A relative path `string` if the URI is in the workspace
+ * @returns `undefined` if the URI is not in the workspace
+ */
+export function relativePath(uri: Uri): string | undefined {
+  if (uri === undefined) {
+    return undefined;
+  }
+  const relativeDirPath = relativeDir(uri);
+  if (!relativeDirPath) {
+    return undefined;
+  }
+  const base = uriUtils.basename(uri);
+  const relativeFilePath = uriUtils.joinPath(Uri.parse(relativeDirPath), base);
+  let result = relativeFilePath.path;
+  if (result.startsWith(path.sep)) {
+    result = result.replace(path.sep, "");
+  }
+  return result;
+}
+
+/**
+ * Returns a list of file path strings, representing the files which are currently
+ * open within the editor.
+ *
+ * @returns An array of relative paths (to the workspace)
+ *          NOTE: paths are not included if they are outside of the workspace
+ */
+export function vscodeOpenFiles(): string[] {
+  // build up a list of open files, relative to the opened workspace folder
+  const openFileList: string[] = [];
+  window.tabGroups.all.forEach((tabGroup) => {
+    tabGroup.tabs.forEach((tab) => {
+      const input = tab.input as TabInputText;
```

Comment:
This initially felt concerning to me, but I tried a host of different types that wouldn't have URIs and since `relativePath` supports `undefined` I couldn't ever get bad behavior.